### PR TITLE
Fix crash when "fix all in solution" is used with "inline declaration".

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/15994

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs [hunk 1]
```diff
@@ -666,8 +666,8 @@ void M()
 {
     void M()
     {
+        // prefix comment
```
Comment (jmarolf) [inline src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs]: Why this change?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs]: Previously we had logic that tried to move the trivia to where we were creating the out-var.  Now we just try to preserve the trivia attaching it to the next statement.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz @srivatsn This fixes several issues with 'Inline Declaration' found by Dave Kean:

https://github.com/dotnet/roslyn/issues/15994
https://github.com/dotnet/roslyn/issues/15992

**Comment by CyrusNajmabadi** [comment]
**Customer scenario**

Customer performs 'fix all in solution' with Inline-Declaration.  If they have any special 'feature' flags set for their project, this will crash.  This is due to us making a new SyntaxTree but not copying those options over.  When we then try to replace the sytnax tree the compiler throws an exception because the options don't match.

**Workarounds, if any**

user has to basically not use 'fix all' or they need to know to remove the feature-flag option from their project.  Neither are discoverable.

**Risk**

Very low.  Fix is localized and barely changes anything.

**Performance impact**

Low/none.

**Is this a regression from a previous update?**

No.  New feature that hit a corner case in the types of proejcts we test on.

**How was the bug found?**

Dave Kean hit it working in the roslyn-project-system project.

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz @srivatsn This fixes several issues with 'Inline Declaration' found by Dave Kean:

#15994
#15992

**Comment by srivatsn** [comment]
Approved pending one more review.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide @Pilchie  I need one more review here.  Thanks!
